From 0d580dd958fcf77065a836fae3c5f045b78293f4 Mon Sep 17 00:00:00 2001
From: SeanChuangWiwynn <Sean_Chuang@wiwynn.com>
Date: Tue, 12 Mar 2024 16:06:58 +0800
Subject: [PATCH 11/12] Workaround for pldmd memory leak

Set memory limit for pldmd, if memeory is over the threshold,
systemd would restart the service.
---
 pldmd/service_files/pldmd.service | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/pldmd/service_files/pldmd.service b/pldmd/service_files/pldmd.service
index 4972822..be22fb4 100644
--- a/pldmd/service_files/pldmd.service
+++ b/pldmd/service_files/pldmd.service
@@ -4,6 +4,9 @@ Wants=xyz.openbmc_project.biosconfig_manager.service
 After=xyz.openbmc_project.biosconfig_manager.service
 
 [Service]
+MemoryAccounting=yes
+MemoryMax=450M
+MemoryHigh=400M
 Restart=always
 Type=dbus
 BusName=xyz.openbmc_project.PLDM
-- 
2.25.1

